### DISCUSSION TITLE
Slayer Task Time Tracking

### DISCUSSION THREAD

**Discussion by red-eye-jedi** [runelite/runelite#16440] [Ideas]
I have been manually tracking the time of each of my slayer tasks completed during wilderness slayer + turael skipping, its nice information and helpful for trying to optimise tasks. If there was a plugin that could start a timer the moment i recieve a task and end the moment i get the task messsage completion with the ability for it to store and remember previous times on tasks that would be the best thing ever. here is a picture of what i have been doing manually 
![image](https://user-images.githubusercontent.com/78451641/228576091-aa8126d3-2ba8-43d5-9c6d-3d731d0b2478.png)

something similar to this would be amazing.

**Comment by red-eye-jedi** [comment]
Bump - im going crazy timing my own tasks

**Comment by king-koda** [comment]
3 years later nothing, bump

**Reply by Avexiis** [reply]
This is something I would actually use and I am kind of surprised nothing exists for it (I checked, too), so, I've decided to give it a shot as a hub plugin. I've gotten the basic implementation of it down, just need to do some testing and probably some bug fixing before I can make a submission, but I'm giving it a try.

**Comment by Avexiis** [comment]
I have created a hub plugin to do this which has been accepted. It is called "Slayer Task Time Tracker." Here's the link: https://runelite.net/plugin-hub/show/slayer-task-time-tracker

Features:
- Begins tracking when you are assigned a new Slayer task. This runs on a lightweight, timestamp based timer.
- Displays the total time and efficient time in the sidebar. Total time is the time between when the task was assigned and when it was finished, including time spent logged out. Efficient time is calculated by how much time you spend in combat with the NPC that is your slayer task.
- Saves past task history for both total and efficient times.
- Highlights tasks that had manual resets red, and displays a tooltip showing how many resets they had.
- Allows for manual deletion of task history, on a per-task basis.
- Shows total combined task completion time at the bottom of the sidebar.
- Offers an overlay that shows both timers, with shift+right click options to configure it.

Here's some pics: 
<img width="242" height="365" alt="image" src="https://github.com/user-attachments/assets/1069905f-dc0f-4080-87af-911189ccec8b" />
<img width="265" height="94" alt="image" src="https://github.com/user-attachments/assets/66b6fea6-4822-4571-aeb5-f25eeb7290dc" />
<img width="241" height="184" alt="image" src="https://github.com/user-attachments/assets/4a7f5a6e-13ff-4ba1-a65d-887f465c1fea" />
